src/hearings/pipes/hearing-answers.pipe.spec.ts

```diff
@@ -56,28 +56,57 @@ describe('HearingAnswersPipe', () => {
-    const expected = cold('(b|)', {b: type});
+    const expected = cold('(b|)', { b: type });
     expect(result$).toBeObservable(expected);
   });
 
   it('should transform case flag', () => {
     const result$ = hearingAnswersPipe.transform(AnswerSource.CASE_FLAGS, of(STATE));
     const caseFlags = '<strong class=\'bold\'>Jane Smith</strong>\n<ul><li>Sign Language Interpreter</li><li>Hearing Loop</li><li>Larger font size</li><li>Reading documents for customer</li><li>Sign Language Interpreter</li><li>Language Interpreter</li></ul><br><strong class=\'bold\'>DWP</strong>\n<ul><li>Physical access and facilities</li></ul><br>';
-    const expected = cold('(b|)', {b: caseFlags});
+    const expected = cold('(b|)', { b: caseFlags });
     expect(result$).toBeObservable(expected);
   });
 
   it('should transform venue', () => {
     const result$ = hearingAnswersPipe.transform(AnswerSource.VENUE, of(STATE));
     const caseFlags = '<ul><li>LIVERPOOL SOCIAL SECURITY AND CHILD SUPPORT TRIBUNAL</li><li>ABERDEEN TRIBUNAL HEARING CENTRE</li></ul>';
-    const expected = cold('(b|)', {b: caseFlags});
+    const expected = cold('(b|)', { b: caseFlags });
     expect(result$).toBeObservable(expected);
   });
 
   it('should transform need welsh', () => {
     const result$ = hearingAnswersPipe.transform(AnswerSource.NEED_WELSH, of(STATE));
     const needWelsh = 'Yes';
-    const expected = cold('(b|)', {b: needWelsh});
+    const expected = cold('(b|)', { b: needWelsh });
+    expect(result$).toBeObservable(expected);
+  });
+
+  it('should transform hearing length', () => {
+    STATE.hearingRequest.hearingRequestMainModel.hearingDetails.duration = 60;
+    const result$ = hearingAnswersPipe.transform(AnswerSource.HEARING_LENGTH, of(STATE));
+    const hearingDuration = '1 hour(s)';
+    const expected = cold('(b|)', { b: hearingDuration });
+    expect(result$).toBeObservable(expected);
+  });
+
+  it('should transform hearing specific date', () => {
+    STATE.hearingRequest.hearingRequestMainModel.hearingDetails.hearingWindow = {
+      hearingWindowDateRange: {
+        hearingWindowStartDateRange: '12-12-2022',
+        hearingWindowEndDateRange: '12-12-2022',
+      },
+      hearingWindowFirstDate: null,
+    };
+    const result$ = hearingAnswersPipe.transform(AnswerSource.HEARING_SPECIFIC_DATE, of(STATE));
+    const hearingDateRange = `${RadioOptions.CHOOSE_DATE_RANGE}<dt class="heading-h3 bottom-0">Earliest hearing date</dt>12 December 2022<dt class="heading-h3 bottom-0">Latest hearing date</dt>12 December 2022`;
+    const expected = cold('(b|)', { b: hearingDateRange });
+    expect(result$).toBeObservable(expected);
+  });
+
+  it('should transform need hearing priority', () => {
+    const result$ = hearingAnswersPipe.transform(AnswerSource.HEARING_PRIORITY, of(STATE));
+    const hearingPriority = 'Standard';
```

Comment:
The priority needs to store the key of the Ref Data not the value of Ref Data in the POST hearing request payload.